jabgui/src/main/java/org/jabref/gui/fieldeditors/LinkedFileViewModel.java

```diff
@@ -268,20 +274,54 @@ private void performRenameWithConflictCheck(String targetFileName) {
         boolean overwriteFile = false;
 
         if (existingFile.isPresent()) {
-            overwriteFile = dialogService.showConfirmationDialogAndWait(
-                    Localization.lang("File exists"),
-                    Localization.lang("'%0' exists. Overwrite file?", targetFileName),
-                    Localization.lang("Overwrite"));
+            // Get existing file path and its directory
+            Path existingFilePath = existingFile.get();
+            Path targetDirectory = existingFilePath.getParent();
+
+            // Suggest a non-conflicting file name
+            String suggestedFileName = FileNameUniqueness.getNonOverWritingFileName(targetDirectory, targetFileName);
+
+            // Define available dialog options
+            ButtonType Replace = new ButtonType(Localization.lang("Replace"), ButtonBar.ButtonData.OTHER);
+            ButtonType keepBoth = new ButtonType(Localization.lang("Keep both"), ButtonBar.ButtonData.OTHER);
+            ButtonType provideAlternative = new ButtonType(Localization.lang("Provide alternative file name"), ButtonBar.ButtonData.OTHER);
+            ButtonType cancel = new ButtonType(Localization.lang("Cancel"), ButtonBar.ButtonData.OTHER);
+
+            // Build the dialog window
+            Dialog<ButtonType> dialog = new Dialog<>();
```

Comment:
Please rework. Use `org.jabref.gui.JabRefDialogService`. Please do not create dialogs on your own. - For example, look at the removed code 😅

MVVM is about testability.

This will also help to test this --> try to craft a test case after you use the JabRefDialogService :)